main.sh

```diff
@@ -167,7 +167,26 @@ case "${OSTYPE}" in
     *) bail "unrecognized OSTYPE '${OSTYPE}'" ;;
 esac
 
-build_options=("--release")
+input_profile=${INPUT_PROFILE:-release}
+
+# There are some special profiles that correspond to different target directory
+# names.
+# See: https://doc.rust-lang.org/nightly/cargo/reference/profiles.html#custom-profiles
+if [[ ${input_profile} = "bench" ]]; then
+    input_profile="release"
+elif [[ ${input_profile} = "dev" ]]; then
+    input_profile="debug"
+elif [[ ${input_profile} = "test" ]]; then
+    input_profile="debug"
+fi
+
+declare -a build_options=()
+
+if [[ -n "${input_profile}" ]]; then
+    build_options+=("--profile" "${input_profile}")
+else
+    build_options+=("--release")
+fi
```

Comment:
`--profile` needs to use the profile name specified by the user, so the order of these two if-blocks needs to be swapped.

Reply:
I split out two variables: `profile` and `profile_directory` to keep things a little clearer.